std/haxe/io/Input.hx

```diff
@@ -77,7 +77,10 @@ class Input {
 			    pos++;
 			    k--;
 			}
-		} catch (eof: haxe.io.Eof){}
+		} catch (eof : Eof){
+			if (k == len)
+				throw eof;
```

Comment:
Some of the overridden versions of readBytes will throw EOF.  One thought here is to have the base version of it throw as well, just so that folks extending it know that it's part of the base behavior.  Haxe doesn't use throwable errors as part of the signature like Java does, so having it in the logic here seems to be the best way to communicate that.

Note that readBytes will eventually hit an eof from its readByte call.  However, it should only throw eof if it wasn't able to read anything at all (k == len).